src/main/java/org/wso2/carbon/inbound/sf/pubsub/SFPubSubInboundFactory.java

```diff
@@ -0,0 +1,239 @@
+import java.util.logging.Logger;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.wso2.carbon.inbound.sf.pubsub.SFConstants.*;
+
+/**
+ * This class is responsible for managing the inbound connection to Salesforce Pub/Sub API.
+ * It handles subscribing to topics and managing subscriptions.
+ */
+public class SFPubSubInboundFactory  extends GenericPollingConsumer {
+    private static final Logger LOGGER = Logger.getLogger(SFPubSubInboundFactory.class.getName());
+    private final String topicName;
+    private int replayPreset;
+    private final int numRequested;
+    private ByteString replayId;
+    private final String server;
+    private final String port;
+    private final String headers;
+    private final String username;
+    private final String password;
+    private final String securityToken;
+    private final boolean tlsEnabled;
+    private PubSubGrpc.PubSubStub asyncStub = null;
+    private ManagedChannel channel = null;
+    private final String subscription_id;
+    private final String developer_name;
+    private final String rpcMethod;
+    private static boolean nextPolling = false;
+    private PubSubGrpc.PubSubBlockingStub blockingStub = null;
+    public SFPubSubInboundFactory(Properties properties, String name, SynapseEnvironment synapseEnvironment, long scanInterval, String injectingSeq, String onErrorSeq, boolean coordination, boolean sequential) {
+        super(properties, name, synapseEnvironment, scanInterval, injectingSeq, onErrorSeq, coordination, sequential);
+        this.topicName = properties.getProperty(TOPIC_NAME);
+        this.replayPreset = getReplayPresetType(properties.getProperty(REPLAY_PRESET));
+        this.numRequested = TypeConverter.convert(properties.getProperty(NUM_REQUESTED), Integer.class);
+        this.replayId = properties.getProperty(REPLAY_ID) == null? null :
+                ByteString.copyFrom(TypeConverter.convert(properties.getProperty(REPLAY_ID), byte[].class));
+        this.server = properties.getProperty(SF_SERVER);
+        this.port = properties.getProperty(PORT_NUMBER);
+        this.headers = properties.getProperty(SF_HEADERS);
+        this.username = properties.getProperty(SF_USERNAME);
+        this.password = properties.getProperty(SF_PASSWORD);
+        this.securityToken = properties.getProperty(SF_SECURITY_TOKEN);
+        this.tlsEnabled = Boolean.parseBoolean(properties.getProperty(TLS_ENABLED));
+        this.rpcMethod = properties.getProperty(RPC_METHOD);
+        this.subscription_id = properties.getProperty(SUBSCRIPTION_ID);
+        this.developer_name = properties.getProperty(DEVELOPER_NAME);
+    }
+
+    @Override
+    public Object poll() {
+        if(nextPolling || (channel != null && (!channel.isTerminated() || !channel.isShutdown()))) {
+            return null;
+        }
+
+        MessageContext msgCtx = createMessageContext();
+        int portInt = Integer.parseInt(port);
+        String target = server + ":" + portInt;
```

Comment:
Do we need to convert the port to an integer?

Reply:
Yeah since the MI UI  pass the value as a string type and no integer data type, therefore we need to convert it to int